### PR TITLE
Guard against nil google doc description

### PR DESCRIPTION
**Pull request by dwarfmondo**
This PR does two things:

* We're seeing a `NoMethodError` exception being raised when `og_data.description` is `nil` 
* Also includes `StandardEmbed` in the google docs engine to retrieve open graph data in a standard way by calling `get_opengraph`

### CODE REVIEW

**Comment by dwarfmondo** [comment]
Hi there, just wanted to follow-up on this to see if anyone has had a chance to look at my changes and if there are any questions. We're currently seeing this in our production environment so getting a fix in-place would great.

**Comment by dwarfmondo** [comment]
@SamSaffron Hi Sam, my apologies for reaching out to you directly (I noticed you've been active on other PRs). I wasn't sure what the protocol is for submitting a PR to this project. Is there anything else I need to do? Thanks!

**Comment by SamSaffron** [comment]
this appears to be a big implementation change, I certainly support fixing the bugs here, but is this all tested? at a minimum a couple of tests should be added that explode on the old code. 

@techAPJ can you review once tests are in?

**Comment by dwarfmondo** [comment]
I was curious about that. The engine has a spec but it's only checking for the type of Google Doc to be correct. I've been in minitest land; let me dust off my rspec skills and add some test coverage to at the very least, test for this particular error.

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /.idea
 .config
 .yardoc
+.ruby-version
```
Comment (CvX) [inline .gitignore:7]: Could you also remove this unrelated change? 🙂

**Comment by techAPJ** [comment]
Hi @dwarfmondo, let us know if you need any help with adding tests.

**Comment by lori7** [comment]
I have a person who stole all of our corporate passwords and we are in a suit and one of the ways to get in is through google and it seemed strange that google warned me on our website the WordPress website password was compromised and when I open the link it goes to a CMS we no longer use.  I thought the source did not look on top of this was strange and he uses sources to make sites link to him to control or defamed me at this time such as my life.  He found anyone name that ends with mine in the US whether they were female and married with his or her new last name and if they had criminal records of any sort, he recreated an HTTP source site and no longer a MyLIfe site after extracting all of the DeVore's that had any criminal records and created an awful discriminating site indicating that I am a rapist, leins on my homes, a repeated under age sexual offender and so forth.  I am Lori DeVore owner of DeVore.com Technologies of 26 years, TestSpots.nets, Cleveland WebWorkscom, Spanish & Global Translator Services, and this man has been to prison for the above and we are in litigation to remove him from my life in Stark County, Canton Ohio and a danger to my life where I am singly living. He is in IT like me and has done prison time and he will go back and in the meantime, he spends his hobbies on attacking my sites in any way he can and it is like StarWars with this man Nathan Deremer. I have a hosting company for 22 years with over 16,000 clients in a very secure data center and keeping up with him attacking me after doing great harm to me and my internal servers and not getting to my clients and I have kept them safe, but he is a 24 hour job for me.  My IT staff helps but they get tired and lazy.

File: lib/onebox/engine/google_docs_onebox.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ module Onebox
   module Engine
     class GoogleDocsOnebox
       include Engine
+      include StandardEmbed
```
Comment (lori7) [inline lib/onebox/engine/google_docs_onebox.rb:7]: https://github.com/discourse/onebox/pull/427/files#diff-f791a38b9399c1968a7f840f68457b47e30bc18b7d8bfe4d5d1c550e69bf41c8
Thank you.

File: lib/onebox/engine/google_docs_onebox.rb [hunk 3]
```diff
@@ -25,10 +26,14 @@ def self.short_types
...
       def data
-        og_data = get_og_data
+        og_data = get_opengraph
+
+        title = og_data.title || "Google #{shorttype.to_s.capitalize}"
+        description = og_data.description.present? ? Onebox::Helpers.truncate(og_data.description, 250) : "This #{shorttype.to_s.chop.capitalize} is private"
+
         result = { link: link,
-                   title: og_data[:title] || "Google #{shorttype.to_s.capitalize}",
-                   description: Onebox::Helpers.truncate(og_data[:description], 250) || "This #{shorttype.to_s.chop.capitalize} is private",
```
Comment (lori7) [inline lib/onebox/engine/google_docs_onebox.rb:31]: Should I change?

File: lib/onebox/engine/google_docs_onebox.rb [hunk 4]
```diff
@@ -45,20 +50,6 @@ def shorttype
       def match
         @match ||= @url.match(@@matcher)
```
Comment (lori7) [inline lib/onebox/engine/google_docs_onebox.rb:51]: Should that be in the string

File: lib/onebox/engine/google_docs_onebox.rb [hunk 5]
```diff
@@ -45,20 +50,6 @@ def shorttype
...
       end
-
-      def get_og_data
-        response = Onebox::Helpers.fetch_response(url, 10) rescue nil
-        html = Nokogiri::HTML(response)
-        og_data = {}
-        html.css('meta').each do |m|
-          if m.attribute('property') && m.attribute('property').to_s.match(/^og:/i)
```
Comment (lori7) [inline lib/onebox/engine/google_docs_onebox.rb:54]: Not comfortable with the NIL?

File: lib/onebox/engine/google_docs_onebox.rb [hunk 6]
```diff
@@ -45,20 +50,6 @@ def shorttype
       def match
         @match ||= @url.match(@@matcher)
       end
-
-      def get_og_data
-        response = Onebox::Helpers.fetch_response(url, 10) rescue nil
```
Comment (lori7) [inline lib/onebox/engine/google_docs_onebox.rb:50]: The response - Onebox::Helpers.fetch_response(url,  10) rescue nil 
should that be in the code?  I am java and I do not see nil ever?
